server/sitemap/sitemap.controller.ts

```diff
@@ -0,0 +1,15 @@
+import { Controller, Get, Request, Response } from "@nestjs/common";
+import { SitemapService } from "./sitemap.service";
+
+@Controller("/")
+export class SitemapController {
+    constructor(private sitemapService: SitemapService) {}
+    @Get("sitemap.xml")
+    async getSitemap(@Response() res, @Request() req) {
+        res.set("Content-Type", "text/xml");
+        const host = req.protocol + "://" + req.get("host");
+        const sitemapXml = await this.sitemapService.getSitemap(host);
+        this.sitemapService.submitSitemap(host);
```

Comment:
Maybe we should have separate endpoints for getting and submitting to google.

It seems that this will create a loop.

Reply:
Also, let's add the submit sitemap endpoint behind a AuthGuard